### ISSUE TITLE
@expo/electron-adapter - "TypeError: this.htmlWebpackPlugin.getHooks is not a function" when running "expo-electron start" using NPM

### ISSUE THREAD

**Issue by RonAmihai** [expo/expo-cli#1451]
I've followed the [documentation guide](https://docs.expo.io/versions/latest/guides/using-electron/) to setup expo-electron project. 

My environment is Windows 10 (1909) with NodeJS v12.14.1 (npm v6.13.4, yarn v1.19.2).
I'm using NPM for all the installations and setup, not yarn.

However, I'm receiving the following output when trying to run app using `yarn expo-electron start`: 

```
yarn run v1.19.2
$ C:\Users\ronam\Downloads\test\prjelec\node_modules\.bin\expo-electron start
┏ Main -----------------------

  Hash: ea452a987b1974f03aa1
  Version: webpack 4.39.0
  Time: 610ms
  Built at: 01/16/2020 09:33:34
    Asset      Size  Chunks             Chunk Names
  main.js  46.6 KiB    main  [emitted]  main
  Entrypoint main = main.js
  [0] multi ./node_modules/electron-webpack/out/electron-main-hmr/main-hmr ./electron/main/index.js 40 bytes {main} [built]
  [./electron/main/index.js] 1.12 KiB {main} [built]
  [./node_modules/@babel/runtime/helpers/interopRequireWildcard.js] 1.15 KiB {main} [built]
  [./node_modules/@babel/runtime/helpers/typeof.js] 449 bytes {main} [built]
  [./node_modules/electron-webpack/out/electron-main-hmr/main-hmr.js] 582 bytes {main} [built]
  [electron] external "electron" 42 bytes {main} [built]
  [electron-webpack/out/electron-main-hmr/HmrClient] external "electron-webpack/out/electron-main-hmr/HmrClient" 42 bytes {main} [built]
  [path] external "path" 42 bytes {main} [built]
  [source-map-support/source-map-support.js] external "source-map-support/source-map-support.js" 42 bytes {main} [built]  [url] external "url" 42 bytes {main} [built]

┗ ----------------------------
┏ Renderer -------------------

  TypeError: this.htmlWebpackPlugin.getHooks is not a function
      at C:\Users\ronam\Downloads\test\prjelec\node_modules\react-dev-utils\InterpolateHtmlPlugin.js:28:10
      at SyncHook.eval (eval at create (C:\Users\ronam\Downloads\test\prjelec\node_modules\tapable\lib\HookCodeFactory.js:19:10), <anonymous>:9:1)
      at SyncHook.lazyCompileHook (C:\Users\ronam\Downloads\test\prjelec\node_modules\tapable\lib\Hook.js:154:20)             at Compiler.newCompilation (C:\Users\ronam\Downloads\test\prjelec\node_modules\webpack\lib\Compiler.js:621:26)          at C:\Users\ronam\Downloads\test\prjelec\node_modules\webpack\lib\Compiler.js:657:29
      at AsyncSeriesHook.eval [as callAsync] (eval at create (C:\Users\ronam\Downloads\test\prjelec\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:6:1)
      at AsyncSeriesHook.lazyCompileHook (C:\Users\ronam\Downloads\test\prjelec\node_modules\tapable\lib\Hook.js:154:20)      at Compiler.compile (C:\Users\ronam\Downloads\test\prjelec\node_modules\webpack\lib\Compiler.js:652:28)
      at C:\Users\ronam\Downloads\test\prjelec\node_modules\webpack\lib\Watching.js:77:18
      at AsyncSeriesHook.eval [as callAsync] (eval at create (C:\Users\ronam\Downloads\test\prjelec\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:24:1)
      at AsyncSeriesHook.lazyCompileHook (C:\Users\ronam\Downloads\test\prjelec\node_modules\tapable\lib\Hook.js:154:20)      at Watching._go (C:\Users\ronam\Downloads\test\prjelec\node_modules\webpack\lib\Watching.js:41:32)
      at C:\Users\ronam\Downloads\test\prjelec\node_modules\webpack\lib\Watching.js:33:9
      at Compiler.readRecords (C:\Users\ronam\Downloads\test\prjelec\node_modules\webpack\lib\Compiler.js:519:11)             at new Watching (C:\Users\ronam\Downloads\test\prjelec\node_modules\webpack\lib\Watching.js:30:17)
      at Compiler.watch (C:\Users\ronam\Downloads\test\prjelec\node_modules\webpack\lib\Compiler.js:244:10)

┗ ----------------------------
Renderer WDS exited with code 1
Error: Renderer WDS exited with code 1
    at ChildProcess.<anonymous> (C:\Users\ronam\Downloads\test\prjelec\node_modules\electron-webpack\src\dev\ChildProcessManager.ts:62:32)
    at ChildProcess.emit (events.js:223:5)
    at maybeClose (internal/child_process.js:1021:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)

```

I have ejected using yarn expo-electron customize in order to check the behavior of htmlWebpackPlugin among the electron setup, but didn't manage to understand how it works out there. 

Maybe you have an idea of possible fix?

**Comment by EvanBacon** [comment]
I am unable to repro using the latest CLI and following the instructions documented:
- `expo init my-project`
- `cd my-project`
- `yarn add -D @expo/electron-adapter`
- `yarn expo-electron`
- `yarn expo-electron start`

Please provide a repro.

**Comment by jimthedev** [comment]
@EvanBacon I tried to repo and cannot reproduce because yarn refuses to install because 7zip dependency has been yanked for being vulnerable. 

<img width="934" alt="Screen Shot 2020-02-11 at 5 18 02 PM" src="https://user-images.githubusercontent.com/108938/74288668-9a22f400-4cf2-11ea-97d6-89c2e698216f.png">

**Comment by efichot** [comment]
Same Issue here: Couldn't find package "7zip@0.0.6" required by "electron-devtools-installer@^2.2.4" on the "npm" registry. with yarn

**Comment by efichot** [comment]
Just need to update the expo-cli to 3.13.1. You can close

**Comment by Knaackee** [comment]
I still get this error with the latest version of the expo cli.
After some debugging I found the problem.

```
$ npm ls html-webpack-plugin

+-- @expo/electron-adapter@0.0.0-alpha.20
| `-- electron-webpack@2.7.4
|   `-- html-webpack-plugin@3.2.0
+-- UNMET PEER DEPENDENCY @expo/webpack-config@0.11.1
| `-- html-webpack-plugin@4.0.0-alpha.2
`-- html-webpack-plugin@4.0.0-alpha.2  extraneous
```

`@expo/electron-adapter` uses `html-webpack-plugin@3.2.0` which does not have the `getHooks `method (available since `html-webpack-plugin@4.0.0`).

So I tried copying the content of the node modules `html-webpack-plugin@4.0.0-alpha.2` over `html-webpack-plugin@3.2.0` and it worked. 

Maybe @EvanBacon could help to fix that.

**Comment by RonAmihai** [comment]
I don't encounter my original issue anymore, closing this issue for now.
